src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -78,3 +68,80 @@
                                vec
                                (sort-by (apply juxt col-key))
                                ds/dataset)))
+
+(defn lag
+  [dataset col-key]
+  (-> dataset
+      (ds/to-map)
+      (col-key)
+      (#(cons 0 %))
+      (butlast)))
+
+(defn calc-proportional-differences
```

Comment:
yes, after coding it- in my mind it's clearer to think about it as a `change in rate from the previous year`.